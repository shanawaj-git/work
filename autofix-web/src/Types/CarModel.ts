import { CarMakerTypeQueryResponse } from "./CarMaker";
export interface CarModelType {
  id: string;
  name: string;
  image?: string;
}

export interface CarModelTypeQueryResponse {
  id: string;
  attributes: {
    image?: string;
    name: string;
    makerId?: {
      data: CarMakerTypeQueryResponse;
    };
  };
}
