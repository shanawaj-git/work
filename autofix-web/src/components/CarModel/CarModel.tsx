import React, { FC } from "react";
import ArrowForwardIcon from "@/assets/arrow-forward-icon.svg";
import { CarModelType } from "../../Types/CarModel";
import { utils } from "@albathanext/design-system";

interface CarModelProps {
  data: CarModelType;
  testId?: string;
  className?: string;
}

const CarModel: FC<CarModelProps> = ({ ...props }: CarModelProps) => {
  const { data, testId, className } = props;
  return (
    <div
      className={"flex p-5 mt-3 items-center justify-between rounded border border-solid".concat(
        " ",
        className || ""
      )}
      data-testid={`CarModel-${testId || data.id}`}
      style={{
        gap: "10px",
        background: utils.colorSelector("c200"),
        borderColor: "rgba(0, 0, 0, 0.08)",
      }}
    >
      <p
        className="font-normal font-[500] text-base leading-6"
        style={{
          color: utils.colorSelector("p700"),
        }}
      >
        {data.name}
      </p>
      <div>
        <img
          className="min-h-3.5 min-w-3 left-1.25 top-4.5 rounded-none"
          src={ArrowForwardIcon}
          alt={ArrowForwardIcon}
        />
      </div>
    </div>
  );
};

export default CarModel;
