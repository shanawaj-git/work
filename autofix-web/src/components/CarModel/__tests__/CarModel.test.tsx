import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CarModel from "../CarModel";

describe("<CarModel />", () => {
  const carModel = {
    id: "1",
    name: "Corolla",
  };

  test("it should mount", () => {
    render(<CarModel data={carModel} />);

    const component = screen.getByTestId(`CarModel-${carModel.id}`);

    expect(component).toBeInTheDocument();
  });

  test("it should match snapshot", () => {
    const { container } = render(<CarModel data={carModel} />);

    expect(container).toMatchSnapshot();
  });
});
