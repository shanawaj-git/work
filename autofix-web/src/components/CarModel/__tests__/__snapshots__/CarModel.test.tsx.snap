// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarModel /> it should match snapshot 1`] = `
<div>
  <div
    class="flex p-5 mt-3 items-center justify-between rounded border border-solid "
    data-testid="CarModel-1"
    style="gap: 10px; background: rgb(255, 255, 255); border-color: rgba(0, 0, 0, 0.08);"
  >
    <p
      class="font-normal font-[500] text-base leading-6"
    >
      Corolla
    </p>
    <div>
      <img
        alt="arrow-forward-icon.svg"
        class="min-h-3.5 min-w-3 left-1.25 top-4.5 rounded-none"
        src="arrow-forward-icon.svg"
      />
    </div>
  </div>
</div>
`;
